shared/src/commonMain/kotlin/org/hyperskill/app/step_quiz/presentation/StepQuizActionDispatcher.kt

```diff
@@ -183,7 +197,12 @@ class StepQuizActionDispatcher(
                     )
                     .fold(
                         onSuccess = { newSubmission ->
+                            if (action.stepRoute is StepRoute.Learn && newSubmission.status == SubmissionStatus.CORRECT) {
+                                doSuspendableAction(Action.CheckTopicCompletion(action.step.topic))
```

Comment:
`doSuspendableAction` should not be called by developer. It's called by framework when handle an `Action` produced by `Reducer`.
`Action.CheckTopicCompletion` is used only inside of Dispatcher, this is why just a plain `suspend` function call is better idea.
For example: `suspend fun checkTopicCompletion(topic: Topic)`. 
`Message.CurrentTopicCompleted` could be produced inside `checkTopicCompletion` function.